plugins/filter_kubernetes/kubernetes.c

```diff
@@ -335,14 +342,49 @@ static int pack_map_content(msgpack_packer *pck, msgpack_sbuffer *sbuf,
         if (log_index == i &&
             (merge_status == MERGE_UNESCAPED || merge_status == MERGE_PARSED)) {
             if (merge_status == MERGE_UNESCAPED || ctx->keep_log == FLB_TRUE) {
-                msgpack_pack_object(pck, k);
+                msgpack_pack_object(pck, *k);
                 msgpack_pack_str(pck, ctx->unesc_buf_len);
                 msgpack_pack_str_body(pck, ctx->unesc_buf, ctx->unesc_buf_len);
             }
         }
         else { /* MERGE_BINARY ? */
-            msgpack_pack_object(pck, k);
-            msgpack_pack_object(pck, v);
+            msgpack_pack_object(pck, *k);
+            msgpack_pack_object(pck, *v);
+        }
+
+        if (k->type == MSGPACK_OBJECT_BIN) {
+            key_ptr  = (char *) k->via.bin.ptr;
+            key_size = k->via.bin.size;
+        }
+        else if (k->type == MSGPACK_OBJECT_STR) {
+            key_ptr  = (char *) k->via.str.ptr;
+            key_size = k->via.str.size;
+        }
+
+        if (key_size < (sizeof(buf_key) - 1)) {
+            memcpy(buf_key, key_ptr, key_size);
+            buf_key[key_size] = '\0';
+            ptr_key = buf_key;
+        }
+        else {
+            /* Long map keys have a performance penalty */
+            ptr_key = flb_malloc(key_size + 1);
```

Comment:
I don't see the corresponding flb_free, you're introducing memory leak here ;-)

Reply:
You're right ! Done !